schema_bump_dry_run_scripts/genes/gene_bump_dry_run.py

```diff
@@ -0,0 +1,227 @@
+        if intersection_genes:
+            deprecated_genes_in_dataset.update(intersection_genes)
+            is_deprecated_genes_found = True
+
+    if is_deprecated_genes_found:
+        dataset_group_key = (*sorted(deprecated_genes_in_dataset), len(dataset_genes_to_compare))
+        # dataset_group_key is a unique key used to identify a group of datasets that have the same set of deprecated
+        # genes and the same number of genes. This is used to group datasets together in the report.
+        if collection_id not in deprecated_datasets:
+            deprecated_datasets[collection_id] = {"dataset_groups": {}}
+        collection = deprecated_datasets[collection_id]
+        if dataset_group_key not in collection["dataset_groups"]:
+            collection["dataset_groups"][dataset_group_key] = {
+                "datasets": [],
+                "num_datasets": 0,
+                "deprecated_genes": deprecated_genes_in_dataset,
+                "num_genes": len(dataset_genes_to_compare),
+            }
+        group = collection["dataset_groups"][dataset_group_key]
+        group["num_datasets"] += 1
+        group["datasets"].append(dataset_id)
+        logger.debug(f"Dataset {dataset_id} has {len(deprecated_genes_in_dataset)} deprecated genes")
+    else:
+        logger.debug(f"Dataset {dataset_id} has no deprecated genes")
+
+    return deprecated_datasets, is_deprecated_genes_found
+
+
+def generate_deprecated_public(base_url: str, diff_map: Dict) -> Dict:
+    """
+    Generate a dictionary of deprecated datasets from public datasets.
+
+    :param base_url: The base URL for fetching public datasets.
+    :type base_url: str
+    :param diff_map: The map of organisms and their deprecated genes.
+    :type diff_map: dict
+
+    :return: A dictionary of public collections with datasets containing deprecated genes.
+    :rtype: dict
+    """
+    public_deprecated = {}
+    for dataset in fetch_public_datasets(base_url):
+        public_deprecated, _ = compare_genes(dataset, diff_map, public_deprecated)
+    for collection in public_deprecated.values():
+        # convert dataset_groups from a dictionary to a list of its values.
+        collection["dataset_groups"] = list(collection["dataset_groups"].values())
+    return public_deprecated
+
+
+def generate_deprecated_private(base_url: str, diff_map: Dict) -> Tuple[Dict, List]:
+    """
+    Generate a dictionary of deprecated private collections its datasets and a list of non-auto-migrated public
+    collections.
+
+    :param base_url: The base URL for fetching private datasets.
+    :type base_url: str
+    :param diff_map: The map of organisms and their deprecated genes.
+    :type diff_map: dict
+    :return: A tuple containing the dictionary of collections with datasets containing deprecated genes and the list
+        of non-auto-migrated public collections.
+    :rtype: tuple
+    """
+    private_deprecated = dict()
+    non_auto_migrated = []
```

Comment:
nit: make set() for constant time lookup of `revision_of not in non_auto_migrated `